### ISSUE TITLE
[ArteTV] Output error messages received from arte.tv API, e.g., due to time restrictions for age restricted content

### ISSUE THREAD

**Issue by auerswal** [yt-dlp/yt-dlp#7782]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm requesting a site-specific feature
- [X] I've verified that I'm running yt-dlp version **2023.07.06** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Germany

### Example URLs

https://www.arte.tv/de/videos/109789-000-A/die-sieben-schwerter/

### Provide a description that is worded well enough to be understood

Due to German broadcasting regulations, some age restricted movies may only be aired during specific times of day. The *ArteTV* online service accordingly restricts access to such material to those specific times of day. For the example URL, this is between 22:00 and 06:00 CEST. This pertains to using the web site with a web browser as well as to using `yt-dlp`.

*During this time window, `yt-dlp` can download such material without any problems.* This is great, thanks! :-)

*Outside of this time windows, `yt-dlp` provides the generic error message `No video formats found!`.* Here might be an opportunity for a slight improvement.

The JSON information received via API (aquired using `yt-dlp --write-pages`) does include pertinent information, especially an `error` object with `code`, `title`, and `message` keys:

```
$ jq .data.attributes.error too_early/109789-000-A_https_-_api.arte.tv_api_player_v2_config_de_109789-000-A.dump 
{
  "code": "ERROR_AGE_RESTRICTION_16",
  "title": "Dieses Programm ist für junge oder empfindsame Zuschauer:innen nicht geeignet.",
  "message": "Es ist nur von 22 bis 6 Uhr verfügbar."
}
```

If the download is attempted during the allowed time window, there is no `error` in the JSON API response:

```
$ jq .data.attributes.error after_dark/109789-000-A_https_-_api.arte.tv_api_player_v2_config_de_109789-000-A.dump 
null
```

It would be nice if the error message provided by *ArteTV* could be printed *in addition* to the generic `No video formats found!` error from `yt-dlp`. This would allow the user to understand that they need to try again later, and might avoid spurious issue reports.

Thanks for your consideration!

(Please note that *ArteTV* in general only provides temporary access to movies etc., so you might need to look for a different example URL when working on this at a later date.)

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', 'https://www.arte.tv/de/videos/109789-000-A/die-sieben-schwerter/']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.07.06 [b532a3481] (zip)
[debug] Python 3.8.10 (CPython x86_64 64bit) - Linux-5.15.0-78-generic-x86_64-with-glibc2.29 (OpenSSL 1.1.1f  31 Mar 2020, glibc 2.31)
[debug] exe versions: ffmpeg 4.2.7, ffprobe 4.2.7
[debug] Optional libraries: certifi-2019.11.28, secretstorage-2.3.1, sqlite3-2.6.0
[debug] Proxy map: {}
[debug] Loaded 1855 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.07.06, Current version: stable@2023.07.06
Current Build Hash: cedf44a9cf3c241520a692c1006d276e6f434d4d945727c47cb1a70d32953f4a
yt-dlp is up to date (stable@2023.07.06)
[ArteTV] Extracting URL: https://www.arte.tv/de/videos/109789-000-A/die-sieben-schwerter/
[ArteTV] 109789-000-A: Downloading JSON metadata
ERROR: [ArteTV] 109789-000-A: No video formats found!; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
Traceback (most recent call last):
  File "/home/auerswald/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 1560, in wrapper
    return func(self, *args, **kwargs)
  File "/home/auerswald/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 1709, in __extract_info
    return self.process_ie_result(ie_result, download, extra_info)
  File "/home/auerswald/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 1768, in process_ie_result
    ie_result = self.process_video_result(ie_result, download=download)
  File "/home/auerswald/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 2732, in process_video_result
    self.raise_no_formats(info_dict)
  File "/home/auerswald/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 1073, in raise_no_formats
    raise ExtractorError(msg, video_id=info['id'], ie=info['extractor'],
yt_dlp.utils.ExtractorError: [ArteTV] 109789-000-A: No video formats found!; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
```

**Comment by dirkf** [comment]
The extractor is already looking at `.data.attributes.rights`, so this is hardly a stretch. It can report the `.title` and `.message` verbatim, or generate an English message based on the `.code`, or some combination of these.

**Comment by gollum-z** [comment]
Another example for this issue: [https://www.arte.tv/de/videos/113529-000-A/heat/](url)
While streaming via `mpv` (using the video URL found by MediathekViewWeb) is possible outside the hours mentioned above, `yt-dlp` says here as follows:
```shell
$ yt-dlp -vU https://www.arte.tv/de/videos/113529-000-A/heat/
[debug] Command-line config: ['-vU', 'https://www.arte.tv/de/videos/113529-000-A/heat/']
[debug] System config "/etc/yt-dlp.conf": ['--limit-rate', '2M', '--no-mtime', '--restrict-filenames', '--no-part', '--prefer-free-formats', '--add-metadata', '--xattrs']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.07.06 [b532a3481] (zip)
[debug] Python 3.8.10 (CPython x86_64 64bit) - Linux-5.15.0-79-generic-x86_64-with-glibc2.29 (OpenSSL 1.1.1f  31 Mar 2020, glibc 2.31)
[debug] exe versions: ffmpeg 4.2.7, ffprobe 4.2.7, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.6.1, certifi-2019.11.28, mutagen-1.44.0, secretstorage-2.3.1, sqlite3-2.6.0, xattr-0.9.6
[debug] Proxy map: {}
[debug] Loaded 1855 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.07.06, Current version: stable@2023.07.06
Current Build Hash: cedf44a9cf3c241520a692c1006d276e6f434d4d945727c47cb1a70d32953f4a
yt-dlp is up to date (stable@2023.07.06)
[ArteTV] Extracting URL: https://www.arte.tv/de/videos/113529-000-A/heat/
[ArteTV] 113529-000-A: Downloading JSON metadata
ERROR: [ArteTV] 113529-000-A: No video formats found!; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
Traceback (most recent call last):
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 1560, in wrapper
    return func(self, *args, **kwargs)
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 1709, in __extract_info
    return self.process_ie_result(ie_result, download, extra_info)
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 1768, in process_ie_result
    ie_result = self.process_video_result(ie_result, download=download)
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 2732, in process_video_result
    self.raise_no_formats(info_dict)
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 1073, in raise_no_formats
    raise ExtractorError(msg, video_id=info['id'], ie=info['extractor'],
yt_dlp.utils.ExtractorError: [ArteTV] 113529-000-A: No video formats found!; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
```
